sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/SuspendingFunCallerDispatcherCheck.kt

```diff
@@ -0,0 +1,38 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.backend.common.descriptors.isSuspend
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.psiUtil.collectDescendantsOfType
+import org.jetbrains.kotlin.resolve.calls.model.DefaultValueArgument
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.FUNS_ACCEPTING_DISPATCHERS
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall as getResolvedCallNative
+
+@Rule(key = "S6311")
+class SuspendingFunCallerDispatcherCheck : CallAbstractCheck() {
+    override fun functionsToVisit() = FUNS_ACCEPTING_DISPATCHERS
+
+    override fun visitFunctionCall(
+        callExpression: KtCallExpression,
+        resolvedCall: ResolvedCall<*>,
+        kotlinFileContext: KotlinFileContext,
+    ) {
+        val bindingContext = kotlinFileContext.bindingContext
+
+        val arguments = resolvedCall.valueArgumentsByIndex
+        if (arguments == null || arguments.isEmpty()) return
+        if (arguments[0] == null || arguments[0] is DefaultValueArgument) return
+
+        /* Last Call Expression is always the call itself, so we drop it */
+        val callExpressions = callExpression.collectDescendantsOfType<KtCallExpression>().dropLast(1)
+        if (callExpressions.isNotEmpty()
+            && callExpressions.all { it.getResolvedCallNative(bindingContext)?.resultingDescriptor?.isSuspend == true }
+        ) {
+            kotlinFileContext.reportIssue(callExpression.calleeExpression!!,
```

Comment:
How about raising the issue on the dispatcher instead of the function name? That would fit a bit better to the message saying "Remove this dispatcher..."